src/styled.js

```diff
@@ -1,29 +1,37 @@
 import { css } from './css';
 
-let h;
-const setPragma = pragma => (h = pragma);
+let h, forwardRef, useTheme;
+const setup = (_h, _forwardRef, _useTheme = () => {}) => {
+    h = _h;
+    forwardRef = _forwardRef;
+    useTheme = _useTheme;
+};
 
 /**
  * Styled function
  * @param {String} tag
  */
 function styled(tag) {
     const _ctx = this || {};
-
     return function() {
         const _args = arguments;
-
-        return function Styled(props) {
-            const _props = (_ctx.p = Object.assign({}, props));
+        function Styled(props, ref) {
+            const theme = useTheme();
+            const _props = (_ctx.p = Object.assign(theme ? { theme } : {}, props));
```

Comment:
Also what do you think?
```suggestion
            const _props = (_ctx.p = Object.assign(useTheme ? useTheme() : {}, props));
```

Reply:
Technically it's possible, but usually `useTheme` just returns `theme` instead of `{ theme }`, e.g. https://github.com/styled-components/styled-components/blob/v5/packages/styled-components/src/hooks/useTheme.js

It'd have to be like this instead,
```
const _props = (_ctx.p = Object.assign(useTheme ? { theme: useTheme() } : {}, props));
```

Let me give it a spin and see how the bundle size looks like, coz with gzip more code != bigger size and vice-versa :grin: